src/components/DateInput/DateInput.test.tsx

```diff
@@ -63,6 +63,18 @@ describe('DateInput', () => {
     expect(container).toMatchSnapshot();
   });
 
+  it('allows selecting a date with time options in 24h mode', async () => {
```

Comment:
can we add a test where we validate that "Hour" values include numbers higher than `12` (while in am/pm it shouldn't)

Reply:
Update: Perhaps the tests found under `src/components/DateRangeInput/DateRangeInput.test.tsx` cover it? Not sure... 😆

Reply:
That's a simple DOM snapshot chacking and the actual integration testing lyes in the `DateRangeInput` suite